pkg/api/api.go

```diff
@@ -68,6 +76,86 @@ func (a *ScratchDataAPIStruct) AuthGetDatabaseID(ctx context.Context) int64 {
 	return ctx.Value("databaseId").(int64)
 }
 
+func (a *ScratchDataAPIStruct) CreateQuery(w http.ResponseWriter, r *http.Request) {
+	apiKey := r.URL.Query().Get("api_key")
+	if apiKey != "local" { // Validate API key
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte("Unauthorized"))
+		return
+	}
+
+	var requestBody struct {
+		Query    string `json:"query"`
+		Duration int    `json:"duration"` // Duration in seconds
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&requestBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Invalid request body"))
+		return
+	}
+
+	// Validate the query
+	if requestBody.Query == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Query cannot be empty"))
+		return
+	}
+
+	// Generate a new UUID
+	queryUUID := uuid.New()
+
+	// Store the query and its expiration time
+	queryExpiration := time.Duration(requestBody.Duration)
+	a.cache.Set(queryUUID.String(), []byte(requestBody.Query), &queryExpiration)
```

Comment:
The cache needs to store both the Query and the `databaseId`. See #116 under "Save query to cache"

> The cache value should **store the database ID and query**. It should also respect the expiration directive.